commitizen/changelog.py

```diff
@@ -281,3 +282,71 @@ def incremental_build(new_content: str, lines: List, metadata: Dict) -> List:
     if not isinstance(latest_version_position, int):
         output_lines.append(new_content)
     return output_lines
+
+
+def get_smart_tag_range(
+    tags: List[GitTag], start: str, end: Optional[str] = None
+) -> List[GitTag]:
+    """Smart because it finds the N+1 tag.
+
+    This is because we need to find until the next tag
+    """
+    accumulator = []
+    keep = False
+    if not end:
+        end = start
+    for index, tag in enumerate(tags):
+        if tag.name == start:
+            keep = True
+        if keep:
+            accumulator.append(tag)
+        if tag.name == end:
+            keep = False
+            try:
+                accumulator.append(tags[index + 1])
+            except IndexError:
+                pass
+            break
+    return accumulator
+
+
+def get_start_and_end_rev(
+    tags: List[GitTag], version: str, tag_format: str
+) -> Tuple[Optional[str], Optional[str]]:
+    """Find the tags for the given version.
+
+    `version` may come in different formats:
+    - `0.1.0..0.4.0`: as a range
+    - `0.3.0`: as a single version
+    """
+    start: Optional[str] = None
+    end: Optional[str] = None
+    try:
+        start, end = version.split("..")
+    except ValueError:
+        end = version
+
+    end_tag = normalize_tag(end, tag_format=tag_format)
+
+    start_tag = None
+    if start:
+        start_tag = normalize_tag(start, tag_format=tag_format)
+
+    tags_range = get_smart_tag_range(tags, start=end_tag, end=start_tag)
+    if not tags_range:
+        return None, None
+
+    start_rev: Optional[str] = tags_range[-1].name
+    end_rev = end_tag
```

Comment:
nitpick: May I know do we need both `end_rev` and `end_tag` variables?

Reply:
I think it's mostly as an alias, to make it more descriptive. The previous line is `start_rev` and to be consistent I added an end_rev. Otherwise `*_rev` and `*_tag` would be mixed. I can change it if you prefer so